### PR TITLE
Document Cloud Viewer Component integration 

### PR DESCRIPTION
**Pull request by vladbailescu**
Component to display PDF's stored in the DAM using the Document Cloud Viewer SDK.

### CODE REVIEW

File: config/src/content/jcr_root/apps/core/wcm/config/com.adobe.cq.wcm.core.components.internal.services.cloudviewer.CloudViewerConfigServiceImpl.config [hunk 1]
```diff
@@ -0,0 +1,16 @@
+# Copyright 2018 Adobe
```
Comment (jckautzmann) [inline config/src/content/jcr_root/apps/core/wcm/config/com.adobe.cq.wcm.core.components.internal.services.cloudviewer.CloudViewerConfigServiceImpl.config:1]: should be 2020

File: content/src/content/jcr_root/apps/core/wcm/components/pdfviewer/v1/pdfviewer/README.md [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+The PDF Viewer component uses the `com.adobe.cq.wcm.core.components.models.PdfViewer` Sling model as its Use-object.
+
+### Context Aware Config
+The PDF Viewer component uses the context aware config `com.adobe.cq.wcm.core.components.internal.services.pdfviewer.PdfViewerCaConfig` with the following properties:
+
+1. `clientId` - **Required** - the client key obtained by registering for the view sdk api - [https://www.adobe.com/go/dcsdks_credentials](https://www.adobe.com/go/dcsdks_credentials)
+2. `reportSuiteId` - Adobe Analytics ID
+
+### Edit Dialog Properties
+The following properties are written to JCR for this Document Cloud Viewer component and are expected to be available as `Resource` properties:
```
Comment (jckautzmann) [inline content/src/content/jcr_root/apps/core/wcm/components/pdfviewer/v1/pdfviewer/README.md:35]: should be renamed:
Document Cloud Viewer component -> PDF Viewer component

File: bundles/core/src/main/java/com/adobe/cq/wcm/core/components/internal/models/v1/PdfViewerImpl.java [hunk 3]
```diff
@@ -0,0 +1,210 @@
...
+        jsonObjectBuilder.add(PN_SHOW_DOWNLOAD_PDF, showDownloadPdf);
+        jsonObjectBuilder.add(PN_SHOW_PRINT_PDF, showPrintPdf);
+
+        return jsonObjectBuilder.build().toString();
+    }
+
+    @Override
+    public String getContainerClass() {
+        if(!StringUtils.isEmpty(type)) {
+            if (type.equals(FULL_WINDOW) && borderless) {
```
Comment (jckautzmann) [inline bundles/core/src/main/java/com/adobe/cq/wcm/core/components/internal/models/v1/PdfViewerImpl.java:194]: I'd use StringUtils.equals() here and below.
`StringUtils.isEmpty(type)` won't be needed anymore.

File: config/src/content/jcr_root/apps/core/wcm/config/com.adobe.cq.wcm.core.components.internal.services.cloudviewer.CloudViewerConfigServiceImpl.config (hunk 1 above)
Comment (vladbailescu) [reply config/src/content/jcr_root/apps/core/wcm/config/com.adobe.cq.wcm.core.components.internal.services.cloudviewer.CloudViewerConfigServiceImpl.config:1]: Good catch. This should have been removed when switching to context-aware configuration.

File: bundles/core/src/main/java/com/adobe/cq/wcm/core/components/internal/models/v1/PdfViewerImpl.java (hunk 3 above)
Comment (vladbailescu) [reply bundles/core/src/main/java/com/adobe/cq/wcm/core/components/internal/models/v1/PdfViewerImpl.java:192]: Done in a691b53

File: content/src/content/jcr_root/apps/core/wcm/components/pdfviewer/v1/pdfviewer/README.md (hunk 2 above)
Comment (vladbailescu) [reply content/src/content/jcr_root/apps/core/wcm/components/pdfviewer/v1/pdfviewer/README.md]: Done in a691b53

File: config/src/content/jcr_root/apps/core/wcm/config/com.adobe.cq.wcm.core.components.internal.services.cloudviewer.CloudViewerConfigServiceImpl.config (hunk 1 above)
Comment (vladbailescu) [reply config/src/content/jcr_root/apps/core/wcm/config/com.adobe.cq.wcm.core.components.internal.services.cloudviewer.CloudViewerConfigServiceImpl.config]: Removed in a691b53

**Pull request merged**